cpp/src/neighbors/ivf_common.cuh

```diff
@@ -148,11 +148,11 @@ __device__ inline auto find_chunk_ix(uint32_t& sample_ix,  // NOLINT
   return ix_min;
 }
 
-template <int BlockDim, typename IdxT>
+template <int BlockDim, typename IdxT, typename DbIdxT>
 __launch_bounds__(BlockDim) RAFT_KERNEL
   postprocess_neighbors_kernel(IdxT* neighbors_out,                // [n_queries, topk]
                                const uint32_t* neighbors_in,       // [n_queries, topk]
-                               const IdxT* const* db_indices,      // [n_clusters][..]
+                               const DbIdxT* const* db_indices,    // [n_clusters][..]
```

Comment:
Have a look at the assignment expression below outside of the PR changeset (lines 173-174):
```C++
neighbors_out[k] =
    valid ? db_indices[clusters_to_probe[chunk_ix]][data_ix] : kOutOfBoundsRecord<IdxT>;
```
`kOutOfBoundsRecord<IdxT>` and `db_indices` here may have different types and I'm not 100% sure how are they unified in the ternary expression. We probably should't change the type of the `kOutOfBoundsRecord`, so I'd suggest to wrap `db_indices` into a `static_cast<IdxT>(...)` to avoid any ambiguity.

Reply:
That is a good idea to make this explicit! I have also added a static check to confirm that it is not a narrowing conversion.